### PR TITLE
refactor(web): mir-501 - refactor the swap component

### PR DESCRIPTION
**Pull request by Drinory**
## Summary

This PR refactors the Swap component to improve maintainability and testability through better separation of concerns:

- Reduced complexity of Swap component: 882 -> 212 lines 
- Extracted custom hooks: `useSwapDataLayer`, `useSwapFormState`, `useSwapValidation`, `useSwapTransaction`, `useSwapLoadingStates`, `useSwapModals`, `useSwapSettings`
- Extracted sub-components: `SwapCard`, `SwapModals`, `SwapActionButton`, `PreviewSummary`, etc.
- Moved fee calculations to data layer, adopted namespaced hook returns for clarity and collision avoidance

No functional changes - only structural improvements.

### 1) Prerequisites
- **Branch:** `drinor/refactor/mir-501-refactor-the-swap-component`
- **Wallet** : one connected wallet is enough, test with at least Fuel Wallet (or the project-standard options)
To cover edge cases quickly, you can either:
1) Temporarily switch to a second account with low ETH (gas) or low source-token balance, or
2) Use the same account and:
    - For insufficient balance, input an amount > your source balance
    - For insufficient gas, temporarily transfer out ETH so the account has near-zero ETH (or use a faucet/test fund on another account)
    
### 2) Happy Path — Basic Swap
1. **Select tokens:** Choose **Source** and **Target** (must be different).
2. **Enter amount:** Type a valid **source amount**.
3. **Quote:** Wait for auto-quote to appear.

**Expected:**
- Quote fetches automatically after input debounce.
- **Action button** shows **“Review”**.
- **Preview card** (after clicking Review) shows:
  - **Rate** (e.g., `1 ETH ≈ 151,696,605.265279980 BERA`)
  - **Routing** (icons + fee tier)
  - **Estimated fees**, **Gas cost**
  - **Price impact** , *percentage*
  - **Warning** if price imact if high/medium
- **Action button** changes to **“Swap”** after Review.

4. **Execute:** Click **Swap**, approve in wallet.
   - Approve → success modal appears with a **“View transaction”** link.
   - Reject → rejection modal appears with appropriate copy.

### 4) Token Flip (Reverse)
1. With a valid quote visible, click the **flip** button between selectors.
2. Wait for quote refresh.

**Expected:**
- Source/Target swap instantly.
- Input amounts map correctly (no stale values).
- New quote is fetched automatically.

---

### 5) Button States & Validation (must match prior behavior)
- **Insufficient balance:** Enter an amount above balance.  
  **Expected:** Button disabled, text **“Insufficient balance”**.
- **Empty inputs:** Clear both fields.  
  **Expected:** Button disabled, text **“Input Amounts”**.
- **Awaiting wallet approval:** Click **Swap** and leave wallet prompt open.  
  **Expected:** Button disabled, text **“Waiting for approval in wallet”**.
- **Insufficient gas:** Use low-gas account.  
  **Expected:** Button disabled, message **“Bridge more ETH to pay for gas”** (or project’s exact gas CTA).

---

### 6) Review Panel Content (no regressions)
With a valid quote and after **Review**:
- **Rate** renders with expected precision.
- **Routing** displays the selected path + fee tier.
- **Estimated fees** and **Gas cost** present and non-zero (when applicable).
- **Price impact** shows correct bucket; if above threshold, a **visible warning** appears.
- Closing and reopening Review preserves state.

---

### 7) Edge & Error Handling
- **Same token selection:** Selecting the same Source/Target should switch the source and target tokens.
- **Target amount editing (reverse calc):** Enter a **target** amount (if supported).  
  **Expected:** Source amount back-calculates; quote refreshes.
- **Rapid edits / debounce:** Type quickly; ensure **no flicker** or inconsistent button states.
- **Wallet disconnect mid-flow:** Disconnect and reconnect.  
  **Expected:** UI returns to a safe state (Review collapses, button state resets).

---

### 8) Modals & UX (from extracted components)
- **Settings modal:** Opens, updates slippage and persists per prior behavior.
- **Token selector modal:** Search & keyboard nav work; selection closes modal and updates quote.
- **Success/Failure modals:** Correct copy; **View transaction** navigates to the correct explorer URL.

---

### 10) Regression Spotlight (due to hook extraction)
- **Loading states:** Spinners/skeletons appear only during fetch.
- **Focus management:** After opening modals, focus is trapped; Escape closes.
- **Performance:** Typing in amount inputs remains smooth; no lag from re-renders.

## Checklist

- [x] I've thoroughly read the latest [contribution guidelines](https://docs.microchain.systems/libraries/contributing/installation).
- [x] I've rebased my branch onto `main` **before** creating this PR.
- [x] My commit messages follow [conventional spec](https://www.conventionalcommits.org/en/)
- [ ] I've added tests to cover my changes (if applicable). 
- [x] I've verified that all new and existing tests have passed locally for mobile, tablet, and desktop screen sizes.
- [ ] My change requires documentation updates.
- [ ] I've updated the documentation accordingly.
- [x] No unrelated changes are included in this PR
- [x] Breaking changes are clearly documented, with migration steps if needed

## Additional Notes

This refactoring maintains 100% backward compatibility. All existing functionality remains unchanged.

Note: Fixed an issue with `ConnectWalletNew` component missing `"use client"` directive that was exposed when moving it to barrel exports.

## Summary by CodeRabbit

* **New Features**
  * Enhanced swap preview with rate, routing, fees, gas estimate and price-impact display.
  * New compact Swap card, dedicated swap action button, and consolidated swap modals (settings, coin selector, success/failure).
  * New hooks powering validation, transaction flow, data layer, form state and unified loading states.

* **Refactor**
  * Swap UI reorganized into modular, hook-driven components; centralized helpers for fee/price calculations and streamlined component exports.

### CODE REVIEW

File: libs/web/src/components/common/Swap/components/SwapActionButton.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <Button
+      disabled={isActionDisabled}
+      onClick={handleSwapClick}
+      size="2xl"
+      className={cn(
+        isActionDisabled &&
+          isRebrandingEnabled &&
+          "bg-accent-primary border-0 text-black hover:bg-accent-primary-1 shadow-none disabled:opacity-100"
+      )}
+    >
```
[the remark this answers is not in the document]
Comment (Drinory) [reply libs/web/src/components/common/Swap/components/SwapActionButton.tsx]: Note: Logged in ([MIR-505](https://charthouse.atlassian.net/browse/MIR-505) ). Unrelated to changes in this PR.

File: libs/web/src/hooks/useSwapTransaction.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+          // Preserve current asset selection, only clear amounts
+          clearAmounts();
+          setReview(false);
+          openSuccess();
+          triggerClassAnimation("dino");
+          await refetchBalances();
+          // TODO: use variable outputs
+          // check this after rebrand
+          await result.waitForResult;
+        }
```
[the remark this answers is not in the document]
Comment (Drinory) [reply libs/web/src/hooks/useSwapTransaction.ts]: This is actually correct as-is. The pattern here is different from the other occurrences in the codebase:

**In `useSwap.ts` (line 118)**, `waitForResult()` is already invoked when the Promise is created:
```typescript
return {
  isStatusPreConfirmationSuccess,
  waitForResult: tx.waitForResult(),  //  Method called here, returns Promise
  id: tx.id,
};
```

**In `useSwapTransaction.ts` (line 104)**, we're awaiting the Promise itself:
```typescript
await result.waitForResult;  //  Awaiting the Promise (no parentheses needed)
```

The type definition (which I just updated) makes this explicit:
```typescript
triggerSwap: (tx: ScriptTransactionRequest) => Promise<{
  isStatusPreConfirmationSuccess: boolean, 
  waitForResult: Promise<TransactionResult<void>>,  //  It's a Promise, not a method
  id: string
} | undefined>;
```

This pattern provides better separation of concerns - the Promise is bound to the transaction at creation time in `useSwap`, then awaited later in `useSwapTransaction`. This is semantically different from the other occurrences where `tx.waitForResult()` is called directly.

Adding `()` here would actually cause a runtime error since `result.waitForResult` is a Promise, not a function.
